tests/bfgs_test.py

```diff
@@ -113,5 +115,24 @@ def test_multiclass_logreg(self, linesearch):
     # Check optimality conditions.
     self.assertLessEqual(info.error, 1e-2)
 
+  def test_n_calls(self):
+    global n_calls
+    n_calls = 0
+
+    n_iter = 4
+
+    def fun(x, *args, **kwargs):  # Rosenbrock function.
+      global n_calls
+      n_calls += 1
+      return sum(100.0*(x[1:] - x[:-1]**2.0)**2.0 + (1 - x[:-1])**2.0)
+
+    with jax.disable_jit(True):
+      x0 = jnp.zeros(2)
+      bfgs = BFGS(fun=fun, tol=1e-7, maxiter=n_iter, implicit_diff=False, stepsize=1e-3)
+      x1, _ = bfgs.run(x0)
+
+    assert n_calls == n_iter + 1
```

Comment:
Use self.assertEqual (the error message will be more intelligible)

Reply:
done